From e950100dff67989996a5cd39f08c93a9eb539c07 Mon Sep 17 00:00:00 2001
From: Chen Baozi <chenbaozi@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:05 +0800
Subject: [PATCH 059/150] dt-bindings: phytium,ixic: Add bindings for Phytium
 IXIC

This patch documents the DT bindings for the Phytium IXIC interrupt
controller which is used to handle PCI legacy interrupt.

Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Change-Id: Ibaa407e804c8cd5c69a90da18896e5f574626183
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 .../interrupt-controller/phytium,ixic.yaml    | 65 +++++++++++++++++++
 MAINTAINERS                                   |  1 +
 2 files changed, 66 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/interrupt-controller/phytium,ixic.yaml

diff --git a/Documentation/devicetree/bindings/interrupt-controller/phytium,ixic.yaml b/Documentation/devicetree/bindings/interrupt-controller/phytium,ixic.yaml
new file mode 100644
index 000000000000..ba7fb2749af4
--- /dev/null
+++ b/Documentation/devicetree/bindings/interrupt-controller/phytium,ixic.yaml
@@ -0,0 +1,65 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/interrupt-controller/phytium,ixic.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Phytium INTx interrupt controller (IXIC)
+
+description: |
+  This is a pseudo interrupt controller to handle PCI legacy interrupt on
+  some Phytium SoCs, which sits between the PCI INTx devices and the GIC
+  and forwards the 4 INTx input signals to 4 adjacent GICv3 SPIs.
+
+
+maintainers:
+  - Chen Baozi <chenbaozi@phytium.com.cn>
+
+allOf:
+  - $ref: /schemas/interrupt-controller.yaml#
+
+properties:
+  compatible:
+    const: phytium,ixic
+
+  reg:
+    description: |
+      Specifies two regions of the register set, which are called
+      'ctr' and 'hpb'
+    minItems: 2
+    maxItems: 2
+
+  interrupt-controller: true
+
+  '#interrupt-cells':
+    description: |
+      Specifies the number of cells needed to encode an interrupt source.
+    const: 3
+
+  intx-spi-base:
+    $ref: /schemas/types.yaml#/definitions/uint32
+    description: |
+      The SPI number of the first SPI of the 4 adjacent ones the IXIC
+      forwards its interrupts to.
+
+required:
+  - compatible
+  - reg
+  - interrupt-controller
+  - '#interrupt-cells'
+  - intx-spi-base
+
+additionalProperties: false
+
+examples:
+  - |
+    ixic: interrupt-controller@29000000 {
+      compatible = "phytium,ixic";
+      reg-names = "ctr", "hpb";
+      reg = <0x0 0x29000000 0x0 0x00060000>,
+            <0x0 0x29100000 0x0 0x00002000>;
+      interrupt-controller;
+      interrupt-parent = <&gic>;
+      #interrupt-cells = <3>;
+      intx-spi-base = <28>;
+    };
diff --git a/MAINTAINERS b/MAINTAINERS
index 78e1a3431c15..46a4a9073ee9 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -2538,6 +2538,7 @@ F:	Documentation/devicetree/bindings/i2c/phytium,i2c.yaml
 F:	Documentation/devicetree/bindings/media/phytium,jpeg.yaml
 F:	Documentation/devicetree/bindings/iio/adc/phytium,adc.yaml
 F:	Documentation/devicetree/bindings/input/phytium,keypad.yaml
+F:	Documentation/devicetree/bindings/interrupt-controller/phytium,ixic.yaml
 F:	Documentation/devicetree/bindings/ipmi/phytium,bt-bmc.yaml
 F:	Documentation/devicetree/bindings/ipmi/phytium,kcs-bmc.yaml
 F:	Documentation/devicetree/bindings/mailbox/phytium,mbox.yaml
-- 
2.47.0

